kokoro/scripts/test/go_test.sh

```diff
@@ -53,6 +53,45 @@ track_flakiness
 # Avoids "fatal: detected dubious ownership in repository" errors on Kokoro containers.
 git config --global --add safe.directory "$(pwd)"
 
+# A helper for parsing YAML files.
+# Ex: VALUE=$(yaml ~/my_yaml_file.yaml "['a_key']")
+function yaml() {
+  python3 -c "import yaml;f=yaml.safe_load(open('$1'))$2;print(','.join(str(i) for i in f) if type(f)==list else f);"
+}
+
+function set_platforms() {
+  # if PLATFORMS is defined, do nothing
+  if [ -n "${PLATFORMS}" ]; then
+    return 0
+  fi
+  # if _LOUHI_TAG_NAME is defined, set TARGET and ARCH env vars by parsing it.
+  # Example value: louhi/2.46.0/shortref/windows/x86_64/start
+  if [ -n "${_LOUHI_TAG_NAME}" ]; then
+    local split_tag=(${_LOUHI_TAG_NAME//\// })
+    TARGET="${split_tag[3]}"
+    ARCH="${split_tag[4]}"
+  fi
+  # if TARGET is not set, return an error
+  if [ -z "${TARGET}" ]; then
+    echo "At least one of TARGET/PLATFORMS must be set." 1>&2
+    return 1
+  fi
+  # if ARCH is not set, return an error
+  if [ -z "${ARCH}" ]; then
+    echo "If TARGET is set, ARCH must be as well." 1>&2
+    return 1
+  fi
+  # At minimum, PLATFORMS will be the distros from "representative" for TARGET/ARCH in projects.yaml.
+  local platforms=$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['representative']")
+  # If not a presubmit job, add the exhaustive list of test distros.
+  if ["${KOKORO_ROOT_JOB_TYPE:-$KOKORO_JOB_TYPE}" != PRESUBMIT_*]; then
+    platforms=${platforms},$(yaml project.yaml "['targets']['${TARGET}']['architectures']['${ARCH}']['exhaustive']") || true
```

Comment:
This is going to double-count things that are both in `representative` and `exhaustive`. They will be tested twice.

Reply:
The yaml is structured without overlaps.